walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,103 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+  /**
+   * Finds available time ranges for all requested attendees to attend the
+   * supplied MeetingRequest, given known events
+   * 
+   * @param events  All known events that will be considered
+   * @param request The meeting request whose attendees must all be free
+   * @return A collection of time ranges where all proposed attendees in @request
+   *         are not attending any known events in @events
+   */
+  public Collection<TimeRange> query(final Collection<Event> events, final MeetingRequest request) {
+    Collection<TimeRange> queryRanges = new ArrayList<>();
+    queryRanges.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true));
+    Map<Event, Map<MeetingRequest, Boolean>> cache = new HashMap<>();
```

Comment:
The query is only related to 1 instance of the request being passed in; what do you try to achieve by having the value be a mapping from MeetingRequest to Boolean?

Reply:
That is actually a great point! I think I overengineered this